client/src/pages/Modals/SocialLink.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import styled, { CSSProp, css } from "styled-components";
+import { FaTimes } from "react-icons/fa";
+import { Flex } from "@convoy-ui";
+import { ILinkTypes, ICON_MAP } from "./SocialLinkInput";
+
+const SocialLinkStyles: Record<ILinkTypes, CSSProp> = {
+  twitter: css`
+    background-color: #00acee;
+  `,
+  github: css`
+    background-color: white;
+    color: ${p => p.theme.colors.dark3};
+  `,
+  instagram: css`
+    background-color: #d72978;
+  `,
+  website: css`
+    background-color: white;
+    color: ${p => p.theme.colors.dark3};
+  `,
+};
+const StyledSocialLink = styled(Flex)<{ type: ILinkTypes }>`
+  padding: ${p => p.theme.space.medium}px;
+  border-radius: ${p => p.theme.radius.small}px;
+  height: 40px;
+
+  a {
+    color: inherit;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    font-size: 14px;
+  }
+
+  ${p => SocialLinkStyles[p.type]}
+`;
+
+function getLocation(href: string) {
```

Comment:
Extract the pathname from URL because i dont wanna show the whole URL in the SocialLink preview.

TODO: refactor.